packages/images-plugin/lib/finder.js

```diff
@@ -1,113 +1,60 @@
-  fixImageTemplateSize: false,
-
-  // whether Appium should ensure that an image template sent in during image
-  // element find should have its scale adjusted to display size so the match
-  // algorithm will not complain.
-  // e.g. iOS has `width=375, height=667` window rect, but its screenshot is
-  //      `width=750 × height=1334` pixels. This setting help to adjust the scale
-  //      if a user use `width=750 × height=1334` pixels's base template image.
-  fixImageTemplateScale: false,
-
-  // Users might have scaled template image to reduce their storage size.
-  // This setting allows users to scale a template image they send to Appium server
-  // so that the Appium server compares the actual scale users originally had.
-  // e.g. If a user has an image of 270 x 32 pixels which was originally 1080 x 126 pixels,
-  //      the user can set {defaultImageTemplateScale: 4.0} to scale the small image
-  //      to the original one so that Appium can compare it as the original one.
-  defaultImageTemplateScale: DEFAULT_TEMPLATE_IMAGE_SCALE,
-
-  // whether Appium should re-check that an image element can be matched
-  // against the current screenshot before clicking it
-  checkForImageElementStaleness: true,
-
-  // whether before clicking on an image element Appium should re-determine the
-  // position of the element on screen
-  autoUpdateImageElementPosition: false,
-
-  // which method to use for tapping by coordinate for image elements. the
-  // options are 'w3c' or 'mjsonwp'
-  imageElementTapStrategy: IMAGE_EL_TAP_STRATEGY_W3C,
-
-  // which method to use to save the matched image area in ImageElement class.
-  // It is used for debugging purpose.
-  getMatchedImageResult: false,
-};
+/**
+ * Checks if one rect fully contains another
+ *
+ * @param {import('@appium/types').Rect} templateRect The bounding rect
+ * @param {import('@appium/types').Rect} rect The rect to be checked for containment
+ * @returns {boolean} True if templateRect contains rect
+ */
+function containsRect(templateRect, rect) {
+  return templateRect.x <= rect.x && templateRect.y <= rect.y
+      && rect.width <= templateRect.x + templateRect.width - rect.x
+      && rect.height <= templateRect.y + templateRect.height - rect.y;
+}
+
 const NO_OCCURRENCES_PATTERN = /Cannot find any occurrences/;
 const CONDITION_UNMET_PATTERN = /Condition unmet/;
 
 
 export default class ImageElementFinder {
-  /** @type {ExternalDriver} */
-  driver;
-
   /** @type {LRU<string,ImageElement>} */
   imgElCache;
 
   /**
-   *
-   * @param {ExternalDriver} driver
-   * @param {number} [maxSize]
+   * @param {number} maxSize
    */
-  constructor(driver, maxSize = MAX_CACHE_SIZE_BYTES) {
-    this.driver = driver;
+  constructor(maxSize = MAX_CACHE_SIZE_BYTES) {
     this.imgElCache = new LRU({
       max: MAX_CACHE_ITEMS,
       maxSize,
-      sizeCalculation: (el) => el.template.length,
+      sizeCalculation: (el) => el.template.length + (el.matchedImage?.length ?? 0),
```

Comment:
ah so here's the source of the memory leak? nice find!

Reply:
no, it is not, but this is anyway more precise than before